src/discussions/comments/data/redux.test.js

```diff
@@ -229,4 +229,142 @@ describe('Comments/Responses data layer tests', () => {
+    allResponses.push(Factory.build('comment', { parent_id: commentId, thread_id: threadId }));
+
+    // Load next comment page
+    axiosMock.onGet(`${commentsApiUrl}${commentId}/`)
+      .reply(200, {
+        results: allResponses.slice(3, 6),
+        pagination: { count: 6, numPages: 2 },
+      });
+    await executeThunk(fetchCommentResponses(commentId, { page: 2 }), store.dispatch, store.getState);
+
+    expect(store.getState().comments.commentsInComments[commentId])
+      .toEqual([
+        'comment-4',
+        'comment-5',
+        'comment-6',
+        'comment-7',
+        // our comment was pushed down
+        'comment-8',
+        // the newer comment is placed correctly
+        'comment-9',
+      ]);
+  });
+
+  test('correctly handles thread comments pagination after posting a new comment', async () => {
+    const threadId = 'test-thread';
+    // New comments are always unendorsed, so we only care about those here
+    const endorsed = EndorsementStatus.UNENDORSED;
+
+    // Build all comments first, so we can paginate over them and they
+    // keep a previous creation timestamp
+    const allComments = Factory.buildList('comment', 4, { thread_id: threadId });
+
+    // Load the first page of comments
+    axiosMock.onGet(commentsApiUrl)
+      .reply(200, {
+        results: allComments.slice(0, 3),
+        pagination: { count: 4, numPages: 2 },
+      });
+    await executeThunk(fetchThreadComments(threadId, { endorsed }), store.dispatch, store.getState);
+
+    expect(store.getState().comments.commentsInThreads[threadId][endorsed])
+      .toEqual([
+        'comment-1',
+        'comment-2',
+        'comment-3',
+      ]);
+
+    // Post new comment
+    const comment = Factory.build('comment', { thread_id: threadId });
+    allComments.push(comment);
+    axiosMock.onPost(commentsApiUrl)
+      .reply(200, comment);
+    await executeThunk(addComment('Test Comment', threadId, null), store.dispatch, store.getState);
+
+    expect(store.getState().comments.commentsInThreads[threadId][endorsed])
+      .toEqual([
+        'comment-1',
+        'comment-2',
+        'comment-3',
+        // our comment was appended to the end
+        'comment-5',
+      ]);
+
+    // Somebody else posted a new response now
+    allComments.push(Factory.build('comment', { thread_id: threadId }));
+
+    // Load next comment page
+    axiosMock.onGet(commentsApiUrl)
+      .reply(200, {
+        results: allComments.slice(3, 6),
+        pagination: { count: 6, numPages: 2 },
+      });
+    await executeThunk(fetchThreadComments(threadId, { endorsed }), store.dispatch, store.getState);
+
+    expect(store.getState().comments.commentsInThreads[threadId][endorsed])
+      .toEqual([
+        'comment-1',
+        'comment-2',
+        'comment-3',
+        'comment-4',
+        // our comment was pushed down
+        'comment-5',
+        // the newer comment is placed correctly
+        'comment-6',
+      ]);
```

Comment:
I think, you can remove all other `expect`'s from the test because they are either redundant, or are being covered by other tests.

Reply:
removed the redundant expects